OctopusDSC/OctopusDSCHelpers.ps1

```diff
@@ -95,16 +95,58 @@ function Write-Log {
     Write-Verbose "[$timestamp] $message"
 }
 
+Function Get-MaskedOutput
+{
+    [CmdletBinding()]
+    param($arguments)
+
+    if(($arguments -match "--masterkey|--password|--license") -gt 0)
+    {
+        # handles cases for --license, --password, --masterkey
+        Write-Verbose "We found a sensitive command line argument. Masking"
+
+        # loop the array, find the culprit indexes
+        for($x=0;$x -lt $arguments.count; $x++)
+        {
+            if(($arguments[$x] -match "--masterKey|--password|--license") -gt 0)
+            {
+                $arguments[$x+1] = $arguments[$x+1] -replace ".", "*"
+            }
+        }
+        $out = $arguments
+    }
+    elseif(($arguments -match "password|pwd") -gt 0)
+    {
+        # handles the password in SQL connection strings
+        Write-Verbose "We found a SQL connection string. Masking"
+
+        # do a regex replace for "password=*[;|$]"
+        $out = $arguments -replace "(password|pwd)=[^;]*", "password=********" 
```

Comment:
greedy-ish...

Reply:
I think

Reply:
Its not a greedy regex, so that part is cool, but it does end up replacing `pwd=` with `password=`...

you want this instead:

```
PS > "a=b;pwd=123;c=d"-replace "(password|pwd)=[^;]*", "`$1=********"   
a=b;pwd=********;c=d
```

Reply:
Cunning. I like it